vana/commands/register.py

```diff
@@ -0,0 +1,132 @@
+# The MIT License (MIT)
+# Copyright © 2024 Corsali, Inc. dba Vana
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import argparse
+import vana
+from vana.commands.base_command import BaseCommand
+from rich.prompt import Prompt, Confirm
+
+
+class RegisterCommand(BaseCommand):
+    """
+    Executes the `register` command to register a validator node on the Vana network using TEE Pool Contract.
+
+    This command allows users to register a validator node with a specified name, URL, and wallet.
+    It interacts with the TeePoolImplementation smart contract to add the validator as a TEE.
+
+    Usage:
+        The command requires specifying the validator name, URL, and wallet name.
+
+    Args:
+        name (str): The name of the validator to register.
+        url (str): The URL of the validator node.
+        wallet (str): The name of the wallet to use for registration.
+
+    Example usage:
+        vanacli register satya --url=https://teenode.com --wallet=validator_4000
+    """
+
+    @staticmethod
+    def run(cli: "vana.cli"):
+        try:
+            chain_manager: "vana.ChainManager" = vana.ChainManager(config=cli.config)
+            wallet = vana.Wallet(config=cli.config)
+
+            # Connect to the contract
+            contract_address = vana.__satori_tee_pool_contract_address
```

Comment:
Can we use [chain_manager.send_transaction](https://github.com/vana-com/vana-framework/blob/main/vana/chain_manager.py#L203) to perform this call, it just needs the contract ABI. We can save some of the duplicated code here.